chain/chain/src/chain.rs

```diff
@@ -4609,6 +4617,18 @@ impl<'a> ChainUpdate<'a> {
                         apply_result.total_balance_burnt,
                     ),
                 );
+                // Right now, we don't implement flat storage for catchup, so we only store
+                // the delta if we are not catching up
+                if !is_catching_up {
+                    if let Some(chain_flat_storage) =
+                        self.runtime_adapter.get_flat_storage_state_for_shard(shard_id)
+                    {
+                        let delta =
+                            FlatStateDelta::from_wrapped_trie_changes(&apply_result.trie_changes);
+                        chain_flat_storage.add_delta(&block_hash, delta);
```

Comment:
:thinking: so this is going to immediately write changes to the database, right? Should it perhaps instead just add them to StoreUpdate, such that the actual update happens when we commit stuff?

Reply:
I get rid of this argument in a following PR.